internal/pkg/template/diff/override.go

```diff
@@ -40,58 +43,185 @@ func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error
+// for correct comparison.
+// E.g. given "!Func: [1,2]" and "Fn::Func: '1,2'", parse assumes that comparing [1,2] with "1,2" produces the desired result.
+// Note that this does not hold for "GetAtt" function: "!GetAtt: [1,2]" and "!GetAtt: 1.2" should be considered the same.
+// parse assumes that from and to are matched by intrinsicFuncMapTagConverter.
+func (*intrinsicFuncMapTagConverter) parse(from, to *yaml.Node, key string, overrider overrider) (diffNode, error) {
 	var diff diffNode
 	var err error
 	if from.Kind == yaml.MappingNode {
 		// The full form mapping node always contain only one child node. The second element in `Content` is the 
 		// value of the child node. Read https://www.efekarakus.com/2020/05/30/deep-dive-go-yaml-cfn.html.
-		diff, err = parse(from.Content[1], stripTag(to), intrinsicFuncFullFormName(from), overrider)
+		diff, err = parse(from.Content[1], stripTag(to), intrinsicFunctions[intrinsicFuncName(from)].fullName, overrider)
 	} else {
-		diff, err = parse(stripTag(from), to.Content[1], intrinsicFuncFullFormName(to), overrider)
+		diff, err = parse(stripTag(from), to.Content[1], intrinsicFunctions[intrinsicFuncName(to)].fullName, overrider)
+	}
+	if diff == nil {
+		return nil, err
+	}
+	return &keyNode{
+		keyValue:   key,
+		childNodes: []diffNode{diff},
+	}, nil
+}
+
+// getAttConverter matches and parses two YAML nodes that calls the intrinsic function "GetAtt".
+// Unlike intrinsicFuncMapTagConverter, getAttConverter does not require "from" and "to" to be written in different form.
+// The input to "GetAtt" could be either a sequence or a scalar. All the followings are valid and should be considered equal.
+// Fn::GetAtt: LogicalID.Att.SubAtt, Fn::GetAtt: [LogicalID, Att.SubAtt], !GetAtt LogicalID.Att.SubAtt, !GetAtt [LogicalID, Att.SubAtt].
+type getAttConverter struct {
+	intrinsicFuncMapTagConverter
+}
+
+// match returns true if both from node and to node are calling the "GetAtt" intrinsic function.
+// "GetAtt" only accepts either sequence or scalar, therefore match returns false if either of from and to has invalid 
+// input node to "GetAtt".
+// Example1: "!GetAtt" and "!GetAtt" returns true.
+// Example2: "!GetAtt" and "Fn::GetAtt" returns true.
+// Example3: "!Ref" and "!GetAtt" returns false.
+// Example4: "!GetAtt [a,b]" and "Fn::GetAtt:a:b" returns false because the input type is wrong.
+func (converter *getAttConverter) match(from, to *yaml.Node, key string, overrider overrider) bool {
+	if !converter.intrinsicFunc.match(from, to, key, overrider) {
+		return false
+	}
+	if intrinsicFuncName(from) != "GetAtt" {
+		return false
+	}
+	fromValue, toValue := from, to
+	if from.Kind == yaml.MappingNode {
+		fromValue = from.Content[1] // A valid full-form intrinsic function always contain a child node.
+	}
+	if to.Kind == yaml.MappingNode {
+		toValue = to.Content[1]
+	}
+	switch {
```

Comment:
is this the same? i.e., `fromValue` and `toValue` both need to be either a scalar or sequence node
```suggestion
	if (fromValue.Kind == yaml.ScalerNode || fromValue.Kind == yaml.SequenceNode) && (toValue.Kind == yaml.ScalerNode || toValue.Kind == yaml.SequenceNode)
```

Reply:
yeah! updated!